src/os/bluestore/BlueStore.cc

```diff
@@ -12196,6 +12207,15 @@ void BlueStore::_kv_sync_thread()
       int r = cct->_conf->bluestore_debug_omit_kv_commit ? 0 : db->submit_transaction_sync(synct);
       ceph_assert(r == 0);
 
+#ifdef HAVE_JAEGER
+      for (auto txc : kv_committing) {
+        if (txc->span) {
+          auto sp = jaeger_tracing::child_span("db sync submit", txc->span);
+          sp->Log({{"kv_committing size", kv_committing.size()}});
```

Comment:
I do not see logs for this span, was the json o/p for recent code?
![image](https://user-images.githubusercontent.com/22215384/121077802-b440e980-c7f5-11eb-89ed-1c30a6d1eee9.png)

Reply:
@ideepika 
The following screenshot and json trace output match the latest code. and span->Log() is used to record metadata.

![new](https://user-images.githubusercontent.com/20939112/121109906-fb9f9800-c83e-11eb-9803-fea855908e21.png)

```json
{
    "data": [
        {
            "traceID": "ea9621b780744d63",
            "spans": [
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "ea9621b780744d63",
                    "flags": 1,
                    "operationName": "ms_fast_dispatch",
                    "references": [],
                    "startTime": 1623116702902716,
                    "duration": 47,
                    "tags": [
                        {
                            "key": "sampler.type",
                            "type": "string",
                            "value": "const"
                        },
                        {
                            "key": "sampler.param",
                            "type": "bool",
                            "value": true
                        },
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "c4daab679da50d5c",
                    "flags": 1,
                    "operationName": "enqueue_op",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae0c47e70d114f1e"
                        }
                    ],
                    "startTime": 1623116702902788,
                    "duration": 17,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [
                        {
                            "timestamp": 1623116702902795,
                            "fields": [
                                {
                                    "key": "cost",
                                    "type": "int64",
                                    "value": 4194304
                                },
                                {
                                    "key": "epoch",
                                    "type": "int64",
                                    "value": 11
                                },
                                {
                                    "key": "owner",
                                    "type": "int64",
                                    "value": 4136
                                },
                                {
                                    "key": "priority",
                                    "type": "int64",
                                    "value": 63
                                },
                                {
                                    "key": "type",
                                    "type": "int64",
                                    "value": 42
                                }
                            ]
                        }
                    ],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "a5f05bb97921e993",
                    "flags": 1,
                    "operationName": "do_op",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae0c47e70d114f1e"
                        }
                    ],
                    "startTime": 1623116702924219,
                    "duration": 8,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "d984dec2e008eaeb",
                    "flags": 1,
                    "operationName": "finish_ctx",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae0c47e70d114f1e"
                        }
                    ],
                    "startTime": 1623116702924430,
                    "duration": 7,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "40acd5324ff0d248",
                    "flags": 1,
                    "operationName": "issue_repop",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae0c47e70d114f1e"
                        }
                    ],
                    "startTime": 1623116702924506,
                    "duration": 8,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "2993dc156056d4cb",
                    "flags": 1,
                    "operationName": "execute_ctx",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae0c47e70d114f1e"
                        }
                    ],
                    "startTime": 1623116702924354,
                    "duration": 8,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "f56772d3c3e0373f",
                    "flags": 1,
                    "operationName": "txc create",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae03b419d2ffcde8"
                        }
                    ],
                    "startTime": 1623116702924672,
                    "duration": 12,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [
                        {
                            "timestamp": 1623116702924677,
                            "fields": [
                                {
                                    "key": "txc seq",
                                    "type": "int64",
                                    "value": 4177
                                }
                            ]
                        }
                    ],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "4d3f80b98c67cfb6",
                    "flags": 1,
                    "operationName": "txc encode finished",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae03b419d2ffcde8"
                        }
                    ],
                    "startTime": 1623116702926167,
                    "duration": 9,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "6651fb5eaa78665d",
                    "flags": 1,
                    "operationName": "eval_repop",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae0c47e70d114f1e"
                        }
                    ],
                    "startTime": 1623116702928922,
                    "duration": 7,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "2e5768b14e85c6d0",
                    "flags": 1,
                    "operationName": "ReplicatedBackend::submit_transaction",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae0c47e70d114f1e"
                        }
                    ],
                    "startTime": 1623116702924574,
                    "duration": 4312,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "1a8655c9d2b21e2d",
                    "flags": 1,
                    "operationName": "txc applied",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae03b419d2ffcde8"
                        }
                    ],
                    "startTime": 1623116702928848,
                    "duration": 10,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "b5fdfe9fe1dd2624",
                    "flags": 1,
                    "operationName": "PGOpItem::run",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae0c47e70d114f1e"
                        }
                    ],
                    "startTime": 1623116702924124,
                    "duration": 4825,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "39e891113f23e713",
                    "flags": 1,
                    "operationName": "do_osd_ops",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae0c47e70d114f1e"
                        }
                    ],
                    "startTime": 1623116702924385,
                    "duration": 5,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "4994eda11c5d6fd5",
                    "flags": 1,
                    "operationName": "do_request",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae0c47e70d114f1e"
                        }
                    ],
                    "startTime": 1623116702924150,
                    "duration": 5,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "4d4889c34c53e202",
                    "flags": 1,
                    "operationName": "aio_wait",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae03b419d2ffcde8"
                        }
                    ],
                    "startTime": 1623116702926219,
                    "duration": 6,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "3c6455fd0c76108e",
                    "flags": 1,
                    "operationName": "io_done",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae03b419d2ffcde8"
                        }
                    ],
                    "startTime": 1623116702934558,
                    "duration": 15,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "ebe6d89444b5aab6",
                    "flags": 1,
                    "operationName": "kv_queued",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae03b419d2ffcde8"
                        }
                    ],
                    "startTime": 1623116702934646,
                    "duration": 8,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "3b06a84b370a865e",
                    "flags": 1,
                    "operationName": "db async submit",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae03b419d2ffcde8"
                        }
                    ],
                    "startTime": 1623116702947916,
                    "duration": 5,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "753fdfcffe7e361a",
                    "flags": 1,
                    "operationName": "kv_submitted",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae03b419d2ffcde8"
                        }
                    ],
                    "startTime": 1623116702948067,
                    "duration": 8,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "03bfdf3723530797",
                    "flags": 1,
                    "operationName": "kv_done",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae03b419d2ffcde8"
                        }
                    ],
                    "startTime": 1623116702960559,
                    "duration": 9,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "368a485c7ec62de2",
                    "flags": 1,
                    "operationName": "finishing",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae03b419d2ffcde8"
                        }
                    ],
                    "startTime": 1623116702960602,
                    "duration": 8,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "33e1bb84f8a5eae2",
                    "flags": 1,
                    "operationName": "done",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae03b419d2ffcde8"
                        }
                    ],
                    "startTime": 1623116702960759,
                    "duration": 9,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "17690422cfbbf1cc",
                    "flags": 1,
                    "operationName": "txc destruct",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae03b419d2ffcde8"
                        }
                    ],
                    "startTime": 1623116702960821,
                    "duration": 11,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "ae03b419d2ffcde8",
                    "flags": 1,
                    "operationName": "BlueStore::queue_transactions",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae0c47e70d114f1e"
                        }
                    ],
                    "startTime": 1623116702924646,
                    "duration": 36231,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [
                        {
                            "timestamp": 1623116702926239,
                            "fields": [
                                {
                                    "key": "pending aios",
                                    "type": "int64",
                                    "value": 64
                                }
                            ]
                        }
                    ],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "7fafa0cf840f1978",
                    "flags": 1,
                    "operationName": "eval_repop",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae0c47e70d114f1e"
                        }
                    ],
                    "startTime": 1623116702977316,
                    "duration": 10,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "ae0c47e70d114f1e",
                    "flags": 1,
                    "operationName": "op-request-created",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ea9621b780744d63"
                        }
                    ],
                    "startTime": 1623116702902756,
                    "duration": 76723,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [],
                    "processID": "p1",
                    "warnings": null
                },
                {
                    "traceID": "ea9621b780744d63",
                    "spanID": "506971f4287602e9",
                    "flags": 1,
                    "operationName": "db sync submit",
                    "references": [
                        {
                            "refType": "CHILD_OF",
                            "traceID": "ea9621b780744d63",
                            "spanID": "ae03b419d2ffcde8"
                        }
                    ],
                    "startTime": 1623116702959626,
                    "duration": 16,
                    "tags": [
                        {
                            "key": "internal.span.format",
                            "type": "string",
                            "value": "proto"
                        }
                    ],
                    "logs": [
                        {
                            "timestamp": 1623116702959635,
                            "fields": [
                                {
                                    "key": "kv_committing size",
                                    "type": "int64",
                                    "value": 5
                                }
                            ]
                        }
                    ],
                    "processID": "p1",
                    "warnings": null
                }
            ],
            "processes": {
                "p1": {
                    "serviceName": "osd-services-reinit",
                    "tags": [
                        {
                            "key": "hostname",
                            "type": "string",
                            "value": "hehe-Latitude-5411"
                        },
                        {
                            "key": "ip",
                            "type": "string",
                            "value": "127.0.1.1"
                        },
                        {
                            "key": "jaeger.version",
                            "type": "string",
                            "value": "C++-0.6.1"
                        }
                    ]
                }
            },
            "warnings": null
        }
    ],
    "total": 0,
    "limit": 0,
    "offset": 0,
    "errors": null
}
```